core/include/scipp/core/element/event_operations.h

```diff
@@ -16,74 +16,93 @@
 
 namespace scipp::core::element::event {
 
-namespace map_in_place_detail {
+constexpr auto get = [](const auto &x, const scipp::index i) {
+  if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(x)>>)
+    return ValueAndVariance{x.value[i], x.variance[i]};
+  else
+    return x[i];
+};
+
+namespace map_detail {
 template <class Coord, class Edge, class Weight>
-using args = std::tuple<span<Weight>, span<const Coord>, span<const Edge>,
-                        span<const Weight>>;
-} // namespace map_in_place_detail
+using args = std::tuple<Coord, span<const Edge>, span<const Weight>>;
+} // namespace map_detail
+
+constexpr auto map =
+    overloaded{element::arg_list<map_detail::args<int64_t, int64_t, double>,
+                                 map_detail::args<int64_t, int64_t, float>,
```

Comment:
Unrelated to this PR but how do you feel about replacing these long lists of possible type with macros?
Some people don't like macros, but in this case, if this sequence is repeated often, it could prevent bugs/copy-paste error by using some predefined macros?

Reply:
I think we can find non-macro solutions (and in fact we do see, e.g., the binary arithmetic kernels). But the main problem is that often these lists are actually different for most operations, so macros do not help.

Reply:
Just a thought - How frequently do we find that the input types are tied (dependent on each other)? Because if they are independent, would it be easier/clearer to provide the types that each of the inputs can have and then use some other mechanism to construct each possible arg_list entry?

Reply:
I think we have that somewhere (iirc it may be used for `mul`), but in this special cases the ties are present and more complicated. In reality we will want to add more type combinations that we have now, but this is a global scipp shortcoming and should be addressed separately.